etl/steps/data/garden/owid/2022-12-19/gdp_last_2000_years.py

```diff
@@ -0,0 +1,139 @@
+    log.info("gdp_last_2000_years.start")
+
+    # Load Maddison garden step
+    ds = Dataset(DATA_DIR / f"garden/ggdc/{VERSION_MADDISON}/ggdc_maddison")
+    df_maddison = ds["maddison_gdp"]
+    df_maddison = pd.DataFrame(df_maddison, dtype=np.float64)
+    df_maddison = df_maddison.reset_index()
+    df_maddison = df_maddison.sort_values(by=["year"]).reset_index(drop=True)
+
+    # Load WDI garden step
+    ds = Dataset(DATA_DIR / f"garden/worldbank_wdi/{VERSION_WDI}/wdi")
+    df_wdi = ds["wdi"]
+    df_wdi = pd.DataFrame(df_wdi, dtype=np.float64)
+    df_wdi = df_wdi.reset_index()
+    df_wdi = df_wdi.sort_values(by=["year"]).reset_index(drop=True)
+
+    # Select GDP and GDP pc in international-$ in 2017 prices
+    df_wdi = df_wdi[["country", "year", "ny_gdp_mktp_pp_kd", "ny_gdp_pcap_pp_kd"]]
+    df_wdi = df_wdi.rename(columns={"ny_gdp_mktp_pp_kd": "gdp", "ny_gdp_pcap_pp_kd": "gdp_per_capita"})
+    # Filter "World" entity
+    df_wdi = df_wdi[df_wdi["country"] == "World"]
+    # Drop empty World GDP estimations
+    df_wdi = df_wdi.dropna().reset_index(drop=True)
+
+    # Load Maddison Dataset 2010 - garden step. It contains global GDP estimations between 1 and 1820
+    ds = Dataset(DATA_DIR / f"garden/ggdc/{VERSION_MADDISON2010}/maddison_database_2010")
+    df_maddison2010 = ds["maddison_database_2010"]
+    df_maddison2010 = pd.DataFrame(df_maddison2010)
+
+    # Select variables to evaluate and reference year to compare Maddison world estimations to World Bank WDI (1990)
+    var_list = ["gdp", "gdp_per_capita"]
+    year_ref = 1990
+
+    # Select only "World" entity in Maddison dataframe
+    df_world = df_maddison[df_maddison["country"] == "World"].copy().reset_index(drop=True)
+    df_world = df_world[df_world["year"] <= year_ref].reset_index(drop=True)
+
+    # Drop population, as it's not needed
+    df_world = df_world.drop(columns=["population"])
+
+    # Merge both Maddison world estimation and WDI world estimation
+    df = pd.merge(df_world, df_wdi, on="year", how="left", suffixes=("_maddworld", "_wdi"))
+
+    for var in var_list:
+        # The reference value for the world aggregation is the 1990 world aggregation value
+        value_ref_countries = df.loc[df["year"] == year_ref, f"{var}_maddworld"].iloc[0]
+        # The scalar is this reference value divided by each each world aggregation
+        df[f"{var}_scalar"] = value_ref_countries / df[f"{var}_maddworld"]
+
+        # The reference value is the 1990 WDI value
+        value_ref = df.loc[df["year"] == year_ref, f"{var}_wdi"].iloc[0]
+        # The rest of the estimated values are the division between the reference value and the scalars
+        df[f"{var}_estimated"] = value_ref / df[f"{var}_scalar"]
+
+        # Rename the estimated to gdp/gdp_per_capita
+        # Round depending on the variable
+        if var == "gdp":
+            df[f"{var}"] = np.where(df[f"{var}_wdi"].isnull(), df[f"{var}_estimated"].round(0), df[f"{var}_wdi"])
```

Comment:
Can you please explain why you round only one of the variables?

Reply:
Very old GDP estimations are subject to uncertainty, so I started by applying `round(0)`, which might end up being `round(-3)` if necessary (we need to agree with Joe, Max et al. about that). It is not necessarily the case for GDP per capita, because it is commonly with decimals and applying the same degree of rounding might turn the estimates meaningless (like having $1000 as world GDP in subsequent years)

Reply:
All measurements have an uncertainty. I would suggest not applying any rounding (except in cases like converting floats to integers).

Reply:
I understand, but these calculations will have a page to discuss the reasons behind that. Very old estimates have a lot of uncertainty and we are playing in top of that. You can see the discussion [here ](https://github.com/owid/owid-issues/issues/379#issuecomment-1363027771) as well

Reply:
If you think rounding is the right thing to do, and given that it's a very significant change in the data, I would make it more transparent. Currently, it's just a `.round` in the middle of a lot of code, easy to miss. You could declare it in one separate line, or at the top of the script (in the docstring), or inside a function (I suggested in another comment to create more functions to do small things).